akka-http-core/src/main/scala/akka/http/javadsl/Http.scala

```diff
@@ -135,12 +154,27 @@ class Http(system: ExtendedActorSystem) extends akka.actor.Extension {
    */
   def bindAndHandle(handler: Flow[HttpRequest, HttpResponse, _],
                     interface: String, port: Int,
+                    connectionContext: ConnectionContext,
+                    materializer: Materializer): Future[ServerBinding] =
+    delegate.bindAndHandle(handler.asInstanceOf[Flow[sm.HttpRequest, sm.HttpResponse, _]].asScala,
+      interface, port, connectionContext.asScala)(materializer)
+      .map(new ServerBinding(_))(ec)
+
+  /**
+   * Convenience method which starts a new HTTP server at the given endpoint and uses the given `handler`
+   * [[Flow]] for processing all incoming connections.
+   *
+   * The number of concurrently accepted connections can be configured by overriding
+   * the `akka.http.server.max-connections` setting.
+   */
+  def bindAndHandle(handler: Flow[HttpRequest, HttpResponse, _],
+                    interface: String, port: Int,
                     settings: ServerSettings,
-                    httpsContext: Optional[HttpsContext],
+                    connectionContext: ConnectionContext,
```

Comment:
why `connectionContext`? If I look at the signature, now it is no longer clear where https settings should go, and on top of that, connection somehow would imply me that `interface` and `port` should belong there.